java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -1141,4 +1157,16 @@ public static FileObject fromUri(String uri) throws MalformedURLException {
         }
         return edits;
     }
+    
+    private static void reportNotificationDone(String s, Object parameter) {
+        if (HOOK_NOTIFICATION != null) {
+            HOOK_NOTIFICATION.accept(s, parameter);
+        }
+    }
+    
+    /**
+     * For testing only; calls that do not return a result should call
+     * this hook, if defined, with the method name and parameter.
+     */
+    static BiConsumer<String, Object> HOOK_NOTIFICATION = null;
```

Comment:
Hard to come up with something better. Opinions, @jlahoda?

The only alternative I can think of is to send some subsequent request (do selection?) and wait for it to finish. Probably too complicated for now.